frontend/src/components/Carousel/index.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { ReactElement, useEffect, useState } from 'react';
+import { CarouselIndicator } from './indicator';
+import { CarouselItem, ICarouselItem } from './item';
+import './index.css';
+
+export * from './item';
+export * from './indicator';
+
+/** 轮播图属性定义 */
+export interface ICarousel {
+  /** 轮播项 */
+  items?: ICarouselItem[];
+  /** 自动滚动间隔 */
+  interval?: number;
+  /** 是否自动滚动 */
+  auto?: boolean;
+  /** 是否显示指示器 */
+  showIndicator?: boolean;
+  children?: ReactElement[];
+  /** 点击轮播项时回调 */
+  onClick?: (item: ICarouselItem, index: number) => void;
+}
+
+/**
+ * 轮播组件
+ * 实现思路：通过CSS translateX实现滚动，通过定时器进行当前展示项变化
+ * @description 轮播图组件可自动或手动点击触发滚动
+ * @param items: ICarouseItem[] 可选，轮播数据集合
+ * @param children: ReactElement[] 可选，组件子项，展示优先级比items高
+ * @param interval: number 可选，滚动间隔，默认为5000ms
+ * @param auto: boolean 可选，是否自动滚动，默认为false
+ * @param showIndicator: boolean 可选，是否展示指示器，默认为true
+ * @param onClick: Function 可选，点击轮播项时触发
+ * @todo：1、后续可尝试通过控制元素展示顺序进行优化，让轮播容器始终保持为200%，同时效果会是无限滚动而不是现在直接回到第一屏
+ * @todo：2、提供更多指示器位置选项，如左侧、右侧、上面等
+ * @todo：3、提供更多轮播项风格选项，如卡片
+ */
+export default function Carousel({
+  items = [],
+  interval = 5000,
+  auto = false,
+  showIndicator = true,
+  onClick,
+  children,
+}: ICarousel) {
+  const [currIndex, setCurrIndex] = useState<number>(0);
+  let timer: NodeJS.Timeout;
+  let count = children && children.length > 0 ? children.length : items.length;
+
+  useEffect(() => {
+    if (!auto) return;
+    timer = setTimeout(() => {
```

Comment:
这里有warning，可否解决一下？
```
src/components/Carousel/index.tsx
  Line 52:13:  Assignments to the 'timer' variable from inside React Hook useEffect will be lost after each render. To preserve the value over time, store it in a useRef Hook and keep the mutable value in the '.current' property. Otherwise, you can move this variable directly inside useEffect  react-hooks/exhaustive-deps

```

Reply:
> 这里有warning，可否解决一下？
> 
> ```
> src/components/Carousel/index.tsx
>   Line 52:13:  Assignments to the 'timer' variable from inside React Hook useEffect will be lost after each render. To preserve the value over time, store it in a useRef Hook and keep the mutable value in the '.current' property. Otherwise, you can move this variable directly inside useEffect  react-hooks/exhaustive-deps
> ```


已修改，修改内容为把timer变量申明放入useEffect中，去除点击指示器时清除timer动作。

![image](https://user-images.githubusercontent.com/7447615/145503117-07980a77-8d6e-49d5-8415-37d3bab95aff.png)